### PR TITLE
Change util test module

### PR DESCRIPTION
**Pull request by bitPogo**
## Description

This separates the `util-test` into 2 modules in order to not pollute projects with coroutines dependencies, if they do not need it.

This depends on #33.

## How is it being implemented?

It simply pulls the blocking test functions out of `util-test`.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have updated the changelog accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: util-coroutine-test/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+                    version {
+                        strictly(Versions.kotlinCoroutines)
+                    }
+                }
+            }
+        }
+
+        val iosTest by getting {
+            dependencies {
+                dependencies {
```
Comment (alessioborraccino) [inline util-coroutine-test/build.gradle.kts:99]: this looks weird

File: util-test/src/commonMain/kotlin/care/data4life/sdk/util/test/CommonResourceLoader.kt [hunk 2]
```diff
@@ -44,6 +44,8 @@ expect class CommonResourceLoader(
 
 // This should be scoped in the ResourceLoader and be private, however this is currently not supported by Kotlin
 internal object CommonPathResolver {
+    const val commonRoot: Path = "src/commonTest/resources"
```
Comment (alessioborraccino) [inline util-test/src/commonMain/kotlin/care/data4life/sdk/util/test/CommonResourceLoader.kt:47]: can this be private?

**Review by alessioborraccino — CHANGES_REQUESTED** [review]

File: util-test/src/commonMain/kotlin/care/data4life/sdk/util/test/CommonResourceLoader.kt (hunk 2 above)
Comment (bitPogo) [reply util-test/src/commonMain/kotlin/care/data4life/sdk/util/test/CommonResourceLoader.kt:47]: Please look at the comment above.

File: util-coroutine-test/build.gradle.kts (hunk 1 above)
Comment (bitPogo) [reply util-coroutine-test/build.gradle.kts:99]: It is weird...why this ever compiled?

File: util-test/src/commonMain/kotlin/care/data4life/sdk/util/test/CommonResourceLoader.kt (hunk 2 above)
Comment (bitPogo) [reply util-test/src/commonMain/kotlin/care/data4life/sdk/util/test/CommonResourceLoader.kt:47]: Or did you mean the prop? This can indeed be private.

Comment (alessioborraccino) [reply util-test/src/commonMain/kotlin/care/data4life/sdk/util/test/CommonResourceLoader.kt:47]: yeah i meant just the const

File: util-coroutine-test/build.gradle.kts (hunk 1 above)
Comment (bitPogo) [reply util-coroutine-test/build.gradle.kts]: done

File: util-test/src/commonMain/kotlin/care/data4life/sdk/util/test/CommonResourceLoader.kt (hunk 2 above)
Comment (bitPogo) [reply util-test/src/commonMain/kotlin/care/data4life/sdk/util/test/CommonResourceLoader.kt]: done

**Review by alessioborraccino — APPROVED** [review]

**Pull request merged**